File: Beecrowd/Beginner/2203 - ~~/2203 - Crowstorm/C++/2203_Crowstorm.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int fiddlesticksX, fiddlesticksY, invaderX, invaderY, invaderSpeed, castingRadius, crowRadius;

    while (cin >> fiddlesticksX >> fiddlesticksY >> invaderX >> invaderY >> invaderSpeed >> castingRadius >> crowRadius) {
        double distance = sqrt(pow(invaderX - fiddlesticksX, 2) + pow(invaderY - fiddlesticksY, 2)) + invaderSpeed * 1.5;
        double range = castingRadius + crowRadius;

        if (distance <= range) {
            cout << "Y" << endl;
        } else cout << "N" << endl;
    }

    return 0;
}

File: Beecrowd/Beginner/2203 - ~~/2203 - Crowstorm/Python/2203_Crowstorm.py
import sys
import math

for line in sys.stdin:
    values = list(map(int, line.split()))
    fiddlesticksX, fiddlesticksY, invaderX, invaderY, invaderSpeed, castingRadius, crowRadius = values

    distance = math.sqrt((invaderX - fiddlesticksX) ** 2 + (invaderY - fiddlesticksY) ** 2) + invaderSpeed * 1.5
    range = castingRadius + crowRadius

    if distance <= range:
        print("Y")
    else:
        print("N")
